rust/cardano-chain-follower/examples/follow_chains.rs

```diff
@@ -318,74 +297,31 @@ async fn follow_for(network: Network, matches: ArgMatches) {
     info!(chain = network.to_string(), "Following Completed.");
 }
 
-/// Helper function for updating the biggest aux data.
-fn update_biggest_aux_data(
-    chain_update: &ChainUpdate, tx_idx: usize, largest_metadata: bool, biggest_aux_data: usize,
-) -> Option<usize> {
-    let raw_size_cip36 = match chain_update
-        .data
-        .txn_raw_metadata(tx_idx, Metadata::cip36::LABEL)
-    {
-        Some(raw) => raw.len(),
-        None => 0,
-    };
-
-    let raw_size_cip509 = match chain_update.data.txn_raw_metadata(tx_idx, cip509::LABEL) {
-        Some(raw) => raw.len(),
-        None => 0,
-    };
-
-    // Get the maximum raw size from both cip36 and cip509
-    let raw_size = raw_size_cip36.max(raw_size_cip509);
-
-    if largest_metadata && raw_size > biggest_aux_data {
-        return Some(raw_size);
-    }
-
-    None
-}
-
-/// Helper function for logging bad CIP36.
-fn log_bad_cip36_info(chain_update: &ChainUpdate, network: Network, tx_idx: usize) {
-    if let Some(decoded_metadata) = chain_update
-        .data
-        .txn_metadata(tx_idx, Metadata::cip36::LABEL)
-    {
-        if let Metadata::DecodedMetadataValues::Cip36(cip36) = &decoded_metadata.value {
-            if (!cip36.signed || !decoded_metadata.report.is_empty())
-                && !decoded_metadata.report.is_empty()
-            {
-                info!(
-                    chain = network.to_string(),
-                    "CIP36 {tx_idx}: {:?}", decoded_metadata
-                );
+/// Function for logging bad CIP36.
+/// Bad CIP36 includes:
+/// - CIP36 that is valid decoded, but have problem.
+/// - CIP36 that is invalid decoded.
+fn log_bad_cip36_info(block: &MultiEraBlock, network: Network) {
+    if let Some(map) = Cip36::cip36_from_block(block, true) {
+        for (key, value) in &map {
+            match value {
+                Ok(value) => {
+                    // Logging the problematic CIP36.
+                    if value.err_report().is_problematic() {
+                        info!(
+                            network = network.to_string(),
+                            "CIP36 valid decoded, but have problem: index {:?}, {}", key, value
+                        );
+                    }
+                },
+                Err(e) => {
+                    warn!("CIP36 decode err {}: ", e);
+                },
             }
         }
     }
 }
 
-/// Helper function for logging CIP509 validation.
-fn log_cip509_info(chain_update: &ChainUpdate, block_num: u64, network: Network, tx_idx: usize) {
```

Comment:
Make this a dummy `log_bad_cip509_info()` with the same parameters as your cip36 version.
And don't do anything but add a `todo` comment.
And call it above where it would be called.